services/api/src/utils/mailer.js

```diff
@@ -1,15 +1,107 @@
 const postmark = require('postmark');
-
 const marked = require('marked');
 const Mustache = require('mustache');
+const { memoize } = require('lodash');
+const fm = require('front-matter');
 const fs = require('fs').promises;
 const path = require('path');
 
 const config = require('@bedrockio/config');
 const { logger } = require('@bedrockio/instrumentation');
 
-const POSTMARK_FROM = config.get('POSTMARK_FROM');
-const ENV_NAME = config.get('ENV_NAME');
+const ENV = config.getAll();
```

Comment:
better to ask for them individually ... i am wondering if you trying get around something here?

Reply:
why is that? I want all configs to be available to the mailers by their exact name in .env... not doing this is unexpected and increases friction IMO

Reply:
The point bedrock/config was to avoid the case where you running in production with env vars that are missing, but only finding out about very late ... as nothing warns you unless you are trying to access the env.

It's not a lot of fiction though? 
We are looking at list of 2-3 env vars no?

Also feel uncomfortable with having all the "secrets" exposed with an bad email.

Reply:
I don't consider secrets to be "exposed" any more than they are to the rest of the code base which also has access.

> running in production with env vars that are missing

I don't think this is true right? As environment variables are inherited they would have to somehow be manually removed, I don't think that's possible.

As a use case we added a "HELP_DESK_URL" to the env. There are untold numbers of variables that might be added so keeping that in sync with the mailer is a lot of friction, yes, especially as the names were different for some vars.

Reply:
`I don't consider secrets to be "exposed" any more than they are to the rest of the code base which also has access.`
My thinking was that you might have dynamic templates in the near future.

```I don't think this is true right? As environment variables are inherited they would have to somehow be manually removed, I don't think that's possible.
```
Thats assuming there is a default, that is not the case always. Specially as we started to move more things into secrets. I am start to think bedrock/config is not solving that problem ... specially how it's being used right now. But let's ignore that, i think i need to rethink bedrock/config. 

```
As a use case we added a "HELP_DESK_URL" to the env. There are untold numbers of variables that might be added so keeping that in sync with the mailer is a lot of friction, yes, especially as the names were different for some vars.
```
Fair enough, i wish we could do like for the web ... we we have a prefix. 

But fine doesn't seems like there is any easy solution. And we are not doing dynamic templates yet so not a worry.

Reply:
> My thinking was that you might have dynamic templates in the near future.

Why would that be an issue?

> Fair enough, i wish we could do like for the web ... we we have a prefix.

We do? I don't see one in master.

Actually I like your first idea better... env configs are ALWAYS public and if they are not then we should put them in secrets instead.

Reply:
thats interesting but it all gets merged... but perhaps there is a way to split them apart

Reply:
ie. anything private would be marked `PRIVATE_` (or something)... that would serve as a signal that it should be inside a k8s secret config (easy to see when in the yaml files) ... however it could also easily allow `config.getPublic()` which could simply be `getAll` but excluding anything marked as private

Reply:
i think there is something there ... but yeah that shouldnt hold this back

Reply:
Just added @bedrockio/config (2.2.1) which now exposes getAllSafe, which returns all keys that has the prefix `SAFE_`

Reply:
ok I don't think this should hold up this PR but fwiw I really think we should flip that. All .env configs should be assumed to be safe by default as we should never be checking in private configs.

Using the `PUBLIC_` scheme will mean we will have to add that prefix everywhere and adds a lot of friction.

Reply:
how much fiction is there really in writing a prefix

Reply:
A lot considering most everything should be exposed by default

Reply:
do you have examples as this does not compute, by far in my experience env vars is for apis that should not be exposed

Reply:
Actually you could be right...hard to know until we actually use it... let's go with this for now 👍